packages/terra-menu/src/_MenuContent.jsx

```diff
@@ -333,7 +353,7 @@ class MenuContent extends React.Component {
     ]);
 
     let header;
-    if (isFullScreen || isSubMenu) {
+    if (isFullScreen || isSubMenu || hasSubMenuItems) {
```

Comment:
```suggestion
    if (isFullScreen || hasHeaderTitle  || isSubMenu) {
        header = this.buildHeader(isFullScreen);
    }
```

Reply:
According to the requirement `add a mandatory top header with a title for any menu that has a drill-in nested menu on the first-tier of the menu (currently only adds the header on the second-tier and higher)` from issue #474 .

Above _suggested changes_ does not meet the issue requirement and fail in a scenario->

When Consumer provides headerTitle to a menu without any sub-menu.
Now `isSubMenu=false`, `hasHeaderTitle=true`
And the header is rendered which does not satisfy the requirement.

The `hasSubMenuItems` check assures to build main-header only when menu has sub-menu items and valid `headerTitle` content.

Reply:
I think if you modify line number : 301 like below. above suggested changes can be included.
```javascript
const hasHeaderTitle = this.props.headerTitle.length > 0 && isSubMenu;
```

Reply:
updated here b1254c430078f5a32a9c1de79d9402410ecc4df2